File: Lab_1/public/backend/validation.php
<?php

require_once "Circle.php";
require_once "Graph.php";
require_once "Square.php";
require_once "Shape.php";
require_once "Triangle.php";

class Validator{
  private Graph $graph;

  public function __construct(){
    $this->graph = new Graph(createShapes());
  }

  public function isPointInShape(float $x, float $y, float $unitR): bool
    {
        return $this->graph->isInGraph($x / $unitR, $y / $unitR);
    }
}

function createShapes(): array
{
  $circle = new Circle(2, 0.5);
  $square = new Square(4, 0.5, -1);
  $triangle = new Triangle(3, -0.5, -1);
  return array($circle, $square, $triangle);
}

File: Lab_3/src/main/java/logic/Validation.java
package logic;
import java.util.List;
import java.util.ArrayList;

public class Validation {
  private final Graph graph;

  public Validation(){
    this.graph = new Graph( createShapes() );
  }

  public boolean isPointInShapes( double x, double y, double unitR ){
    return this.graph.isInGraph( x / unitR, y / unitR );
  }

  private ArrayList<ShapeInterface> createShapes() {
    Circle circle = new Circle(4, 1);
    Square square = new Square(2, -1, 0.5);
    Triangle triangle = new Triangle(1, 1, 0.5);
    return new ArrayList<>( List.of(circle, square, triangle) );
  }
}